server/src/main/java/org/opensearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -171,6 +176,11 @@ public NodeStats(StreamInput in) throws IOException {
         } else {
             weightedRoutingStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
I think we'll want to gate these writes behind the feature flag as well. We want to be able to make backwards-incompatible changes to this feature (like maybe remove this or change the structure) and not impact any instances that have not enabled the experimental feature flag. The only way to do that (I think) is to make sure we don't write anything over the wire unless the feature flag is turned on.